plugins/storage/fileshare/storagefsvm/src/main/resources/conf/fsvm-init.yml

```diff
@@ -0,0 +1,196 @@
+          mkfs.ext4 "$PARTITION"
+          log "Formatted Partition $PARTITION with EXT4 Filesystem."
+        else
+          log "Invalid filesystem type specified. Use 'xfs' or 'ext4'."
+          exit 1
+        fi
+      else
+        log "Partition $PARTITION already has a filesystem of type $FS_TYPE. Skipping format."
+      fi
+
+      FS_INFO=$(blkid "$PARTITION")
+      UUID=$(echo "$FS_INFO" | grep -oP "UUID=\"\K[^\"]+")
+      TYPE=$(echo "$FS_INFO" | grep -oP "TYPE=\"\K[^\"]+")
+
+      if [ -z "$UUID" ] || [ -z "$TYPE" ]; then
+        log "Failed to retrieve UUID or TYPE for $PARTITION"
+        exit 1
+      fi
+
+      echo "UUID=$UUID $MOUNT_POINT $TYPE defaults 0 2" >> /etc/fstab
+      log "Added fstab entry."
+
+      mount -a
+
+      if mountpoint -q "$MOUNT_POINT"; then
+        log "$PARTITION is successfully mounted on $MOUNT_POINT"
+      else
+        log "Failed to mount $PARTITION on $MOUNT_POINT"
+        exit 1
+      fi
+
+      if [ ! -d "$EXPORT_DIR" ]; then
+        log "Creating export directory..."
+        mkdir -p "$EXPORT_DIR"
+        chown nobody:nogroup "$EXPORT_DIR"
+        chmod 777 "$EXPORT_DIR"
+      fi
+
+      log "Configuring NFS export..."
+      EXPORT_ENTRY="$EXPORT_DIR *($PERMISSIONS)"
+      if ! grep -qF "$EXPORT_ENTRY" /etc/exports; then
+        echo "$EXPORT_ENTRY" | tee -a /etc/exports > /dev/null
+      fi
+      exportfs -ra
+
+      log "Enable and restart NFS server..."
+      systemctl enable nfs-kernel-server
+      systemctl restart nfs-kernel-server
+
+      log "NFS share created successfully."
+      log "Finished running add script."
+
+  - path: /usr/local/bin/resize-partition
+    permissions: '0700'
+    owner: root:root
+    content: |
+      #!/bin/bash -e
+
+      LOG_FILE="/var/log/userdata.log"
+      log() {
+        echo "$(date '+%Y-%m-%d %H:%M:%S') - $1" >> $LOG_FILE
+      }
+      log "Running resize script."
+      PARTITION="$1"
+
+      FILESYSTEM={{ fsvm.filesystem }}
+      if [ "$FILESYSTEM" == "xfs" ]; then
+        xfs_growfs "$PARTITION"
+      elif [ "$FILESYSTEM" == "ext4" ]; then
+        resize2fs "$PARTITION"
+      else
+        log "Invalid filesystem type specified. Use 'xfs' or 'ext4'."
+        exit 1
+      fi
+
+      log "Finished running resize script."
+
+  - path: /etc/systemd/system/cloud-dhclient@.service
+    permissions: '0700'
+    owner: root:root
+    content: |
+      [Unit]
+      Description=CloudStack service to start dhclient
+
+      [Install]
+      WantedBy=multi-user.target
+
+      [Service]
+      Type=simple
+      ExecStart=/usr/sbin/dhclient %I
+      Restart=on-failure
+
+  - path: /usr/local/bin/fsvm-setup
+    permissions: '0700'
+    owner: root:root
+    content: |
+      #!/bin/bash -e
+
+      LOG_FILE="/var/log/userdata.log"
+      log() {
+        echo "$(date '+%Y-%m-%d %H:%M:%S') - $1" >> $LOG_FILE
+      }
+
+      HYPERVISOR_INFO=$(dmesg | grep -i hypervisor)
```

Comment:
hypervisor can be set in a similar way as `{{ fsvm.filesystem }}`

Reply:
But how would we know what is the hypervisor type before calling `createAdvancedVirtualMachine` in `createFileShareVM` as host won't be decided yet. If we have multiple clusters with different hypervisors, VM can be deployed on any of them?

Reply:
@abh1sar based on the code I see, template (and in turn hypervisor) is already selected when `createAdvancedVirtualMachine` is passed so would it make sense to pass same hypervisor type?